api/main.py

```diff
@@ -59,6 +66,54 @@ def require(keys: list[str], type = str) -> str:
     keys = "' or '".join(keys)
     flask.abort(400, f"Missing required parameter '{keys}'")
 
+@app.route('/calendar', methods=['GET'])
+def calendar() -> str:
+    count = flask.request.args.get('n', None, int)
+    SCOPES = ["https://www.googleapis.com/auth/calendar.readonly"]
+    creds = None
+    if os.path.exists("token.json"):
+        creds = Credentials.from_authorized_user_file("token.json", SCOPES)
+    # If there are no (valid) credentials available, let the user log in.
+    if not creds or not creds.valid:
```

Comment:
This segment in here is almost identical to the __authenticate code in the gmail object in the library. Let's leave it for now, I'll refactor to combine. I actually like the use of json better than the pickle code that I have already.

Reply:
Generally speaking, it's a good idea to 1. keep all methods short (at 40 lines this on is getting long) and 2. keep logic out of the API/front end layer.

Notice how you made inline comments above? Those should be triggers that you really want the code to live somewhere more modular and targeted at a single behavior.